172.md

```diff
@@ -0,0 +1,101 @@
+
+    // moderators
+    ["p", "<32-bytes hex of a pubkey1>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey2>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey3>", "<optional recommended relay URL>", "moderator"],
+	
+    // relays used by the community (w/optional marker)
+    ["relay", "<relay hosting author kind 0>", "author"],
+    ["relay", "<relay where to send and receive requests>", "requests"], 
+    ["relay", "<relay where to send and receive approvals>", "approvals"],
+    ["relay", "<relay where to post requests to and fetch approvals from>"]
+  ]
+}
+```
+
+# New Post Request
+
+Any Nostr event can be a post request. Clients MUST add the community's `a` tag to the new post event in order to be presented for the moderator's approval.
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": 1,
+  "tags": [
+    ["a", "34550:<Community event author pubkey>:<d-identifier of the community>", "<Optional relay url>"],
+  ], 
+  "content": "<My content>"
+}
+```
+
+Community management clients MAY filter all mentions to a given `kind:34550` event and request moderators to approve each submission. Moderators MAY delete his/her approval of a post at any time using event deletions (See [NIP-09](09.md)).
+
+# Post Approval by moderators
+
+The post-approval event MUST include `a` tags of the communities the moderator is posting into (one or more), the `e` tag of the post and `p` tag of the author of the post (for approval notificaitons). The event SHOULD also include the stringified `post request` event inside the `.content` ([NIP-18-style](18.md)) and a `k` tag with the original post's event kind to allow filtering of approved posts by kind.
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": "4550",
+  "tags": [
+    ["a", "34550:<Community event author pubkey>:<d-identifier of the community>", "<Optional relay url>"],
+    ["e", "<Post Request ID>", "<Optional relay url>"],
+    ["p", "<Post Request Author ID>", "<Optional relay url>"],
+    ["k", "<New Post Request kind>"],
+  ], 
+  "content": "<New Post Request JSON>"
+}
+```
+
+It's recommended that multiple moderators approve posts to avoid deleting them from the community when a moderator is removed from the owner's list. In case the full list of moderators must be rotated, the new moderator set must sign new approvals for posts in the past or the community will restart. The owner can also periodically copy and re-sign of each moderator's approval events to make sure posts don't dissapear with moderators. 
```

Comment:
How about making the moderator list append only and give each moderator a expiration date? Then you can remove somebody starting last week without affecting all his approvals. As these approvals are even bigger than whatever they are approving, duplicating them should not be overly encouraged.

Reply:
I don't think it works because the expired moderator can post in the past. I don't think that should be allowed once a moderator is kicked out.